### DISCUSSION TITLE
binding model array attributes

### DISCUSSION THREAD

**Discussion by vg0426** [livewire/livewire#4629] [1. Help]
hi guys, i have a model that i am binding in a livewire component, this model has a json attribute that is been cast to array in the model definition, i can bind to input text fields all model attributes including the attributes inside the array, my problem is now that i would like to delete and add more attributes inside the array, i tried doing an unset inside the updated function but i am getting error "Indirect modification of overloaded property", is there a way to achieve this?

**Comment by silnarm** [comment]
Sorry to say, but I don't think you are going to be able to do this currently.

The actual array (if there is one) is hidden away somewhere in the Model, and can only be accessed via magic __get(), which gives you a copy. This is where your indirect modification errors come from, in trying to set something on (or unset) an array that you can only get a copy of.

You can actually fix the adding/removing problem that you mentioned, but taking the array copy you are given and manipulating it, then setting the whole property back to the amended array, so instead of a nice simple unset() you would have:
```
        $arr = $this->foo->bar;
        unset($arr['baz']);
        $this->foo->bar = $arr;
```
and similarly for adding.

The bigger issue is you will get the same errors when trying to update individual values within the array, and I can't give you a workaround for that, I think it will require changes to the way Livewire deals with models.

**Comment by joshhanley** [comment]
@vg0426 yeah @silnarm is correct, this is more of a Laravel model issue than a Livewire one.

There are ways to setup getters and setters in your models to enable binding to your array properties, but I find that can be troublesome.

I find it much easier to create a public property for your array and getting the data from the model in your `mount()` method and assigning it to the property. Then when you are ready to save your model, re-assign your array as described by @silnarm and then save the model.

Hope this helps!
